k9mail-library/src/test/java/com/fsck/k9/mail/filter/FixedLengthInputStreamTest.java

```diff
@@ -0,0 +1,230 @@
+                "uexijkkVShZXIvcYgZdljYyIOQCrlLtHcGMZNJFhCcWJqcriZLRZctrrESxxKZWkGeVcKLFMTHcMwidbiAchSpjQBvCwMXpAlpSt" +
+                "seWUSbdfGQqdadWczKVrEYaiDrhNeepPFDllynREhXUBspdOsRPkVMVaNbcmJEwQkoMeWlsfvRpdAuBZYTzfKVDKQGDoDVDhJyeA" +
+                "BnzjGEEoiGrOiLGoViHpiLCBOLBiLCtfxSWKYuKuixEAJOBiMZFbHGBVDJmUNoeXfxXKNabolkDTCwcPohDrsvvinNCHDSOYnfXp" +
+                "olFRYTClufMizjJMzgBYApgrhOFZpAmmJaTadOgIVVafWvEFKOefUILvQNLaxdWuDVmzOxsSIZiWvYWbpfpIqEMgmfaUpHVdxHOu" +
+                "xbYPqezLbQPjXwULNznEyHYpyCpAFOawJpMmHfluqsSNMzNVlyafSPLDywjvEMIqcYWSmDdcfUQMHbzkECgkrYJuinAWYUPjoJRZ" +
+                "QkxcfkTlhhoFnmbZyvLSrIVcNYdLIDUsqCLQAXEJIJVegUJPfzssOmWHDAPZyLtylUUgFyxvwxLFgAnehydWWXnXIVmnqZEeXeAh" +
+                "cANnXpHqMcAtNXgmBkELyqrYemyPsnlvWMNFJsjJcYNoaAwFiAKVxebtwLLvodQFtuJxasxTcQvsiwSRjWanjWnMDoJKoQtwOcCK" +
+                "mKxHxyUuggufOjgJGvUoUooMBpMXDGrNohNzJbQmnCuFVHxulXEircCYtezkVeEkJKFRDXUlwvOSrfhTmnoytjeQbBcIaRZYsRAn" +
+                "isNlMSHqEXQIDirVGicrcOUxQzYhZlnHqwbKslxGkAjIPesZeQKSgbDMnFaAmVLOOrONYuszgMCNCvNGYJFOWGpZUmNvWvTWFoEX" +
+                "QrPSvEmhxyYmFuIfJppvJSDlggOwcXYiXdtgoFfRZyTQbQmdnhoVHvQffkcMSqHDGRpwSLRbqDWwHJOkVJSMFzHdttJwyVpWuTDa" +
+                "uFXOuHfpoXkHStuKkreyWLsasXTRVchqRaEYjUmcTuNZYlcVPFezKqIaDiRcRxuuYFtMzJqOIwMNmITofhdQiXTIfUHGfsYnPYgM" +
+                "wpVdzGoyIrSSKJjWMnGdhmJHwHoFCqNtXOnbltOMnlCuzBsNUIlrKgfHRTAzAoFAlMvxgIrGffqCLtpAdbCbxzxMuivOctNDhuXy" +
+                "iTuSYUvHnxZxklJDuMbMeGqQddtmAMVAhGNPVfdBaNQiDNxrRqLqFuYMjkDwekzDSgkFbCEiHPwmAqAuoRDrtnwXQfkhyKLNlhRE" +
+                "ZxDybKsCrrQPcrNFKskrvceHPGjvyBtVgaHTSmtzdMSVwNyMIlmsMIgZqpVCNVBmwPVYkEfuenVdGeQcSdGSEMuJGOdWXyMbVQiN" +
+                "hKKwnoffMoKJEbfgSLOMsSVUwEBRGFBPFhhsKLZrmPIdSMKwxhPOvzGywVCamswfIspSUHXyvtQrWGNUkyVyYWwjwumRSVfDFepT" +
+                "JnNHLfbAdkCfHsEbUQCHBlSlgpYExOjYyuYbLjCjBYdNXcfdShRRGhzcdsxmsbAnHunqzIeVsOrlJRfdhaDYtmeIOUBCqFZBpLXT" +
+                "jkvAhXYTjLCmcKebgCUFWCLgYnfxYIrTnbLuEyNPeIlyNIWefQrpYKHtEaUVESWrHYFWGHObaKnpspDilNhkfrWdFXUyZEgkbBPC" +
+                "DSUAsUKPpFNyrWPZCPHzVARrspCJbXNCkotJGtgrDdLMMGIPLEPirRZzRxYgDiefAsUvCTnQXGHTOXTCRgjZCegPBxXTblJVFepV" +
+                "YzPbwggNJDRgrHbtLvEQilSMnHSmeJahMUjJXXHFwZXQlMRVrPolpogKooCqIYErEiNnaWrbOaQhLRVxGCGHVnSaHyqaFQwdaSUF" +
+                "JRjJWyDvRjwIAVYTEghKdtdHnATaBxrExrpBmIdbZciYLtZszlxmziScLwdJWaXVycSAdubKkZGlozHnjBVRqllIzFufoSLgxExE";
+
+        STR_LENGTH = string.length();
+    }
+
+    // Test the cases that fixed length is smaller than the length of bytes array
+
+    @Test
+    public void testLongInput() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT1);
+
+        final int LENGTH = 10000;   // LENGTH > LENGTH_LIMIT1
+        byte[] bytes = new byte[LENGTH_LIMIT1];
+
+        Assert.assertEquals(fixedLengthInputStream.read(bytes, 0, LENGTH), LENGTH_LIMIT1);
```

Comment:
This gets the order of arguments wrong. It's `assertEquals(expected, actual)`.

Also, `InputStream.read(byte[], int, int)` does not guarantee to read all of the requested bytes. So failing this assertion wouldn't mean the implementation is broken.